### PR TITLE
[Geant4] Upgrade version to 11.2.1 and add windows platform

### PR DESCRIPTION
**Pull request by peremato**
- Needed to add patch file for windows

### CODE REVIEW

File: G/Geant4/bundled/patches/windows.patch [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+--- a/source/processes/hadronic/models/lend/include/nf_specialFunctions.h
++++ b/source/processes/hadronic/models/lend/include/nf_specialFunctions.h
+@@ -11,7 +11,7 @@
+ 
+ #include "nf_utilities.h"
+ 
+-#ifdef WIN32
++#ifdef _WIN32
+ #define M_PI 3.141592653589793238463
+ #endif
```
Comment (giordano) [inline G/Geant4/bundled/patches/windows.patch:77]: Side note, the way to get `M_PI` on Windows is to define `_USE_MATH_DEFINES` before loading `math.h`/`cmath`: https://learn.microsoft.com/en-us/cpp/c-runtime-library/math-constants?view=msvc-170

File: G/Geant4/bundled/patches/windows.patch [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+index afa8a06987..e6e8edfe26 100644
+--- a/source/processes/hadronic/models/lend/src/xDataTOM_importXML.cc
++++ b/source/processes/hadronic/models/lend/src/xDataTOM_importXML.cc
+@@ -11,8 +11,8 @@
+ #include <fcntl.h>
+ #include <errno.h>
+ 
+-#if defined(WIN32) || defined(__MINGW32__)
+-#include <BaseTsd.h>
++#if defined(_WIN32) || defined(__MINGW32__)
```
Comment (giordano) [inline G/Geant4/bundled/patches/windows.patch:114]: This is equivalent to
```suggestion
+#if defined(_WIN32)
```
no?

File: G/Geant4/bundled/patches/windows.patch [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+ #endif
+ 
+diff --git a/source/global/management/src/G4Threading.cc b/source/global/management/src/G4Threading.cc
+index 3c6bd77b14..fe09e672c7 100644
+--- a/source/global/management/src/G4Threading.cc
++++ b/source/global/management/src/G4Threading.cc
+@@ -35,7 +35,7 @@
+ #include "globals.hh"
+ 
+ #if defined(WIN32) || defined(__MINGW32__)
```
Comment (giordano) [inline G/Geant4/bundled/patches/windows.patch:34]: Also here
```c
 #if defined(_WIN32)
```
is more sensible

File: G/Geant4/bundled/patches/windows.patch
Comment (giordano) [inline G/Geant4/bundled/patches/windows.patch:1]: It'd be great to upstream these changes :slightly_smiling_face:

File: G/Geant4/bundled/patches/windows.patch (hunk 1 above)
Comment (peremato) [reply G/Geant4/bundled/patches/windows.patch:77]: Yes I know, I used in one occasion.  But I wanted to do minimal patch. They define this several times.

File: G/Geant4/build_tarballs.jl [hunk 4]
```diff
@@ -101,4 +109,4 @@ dependencies = [
 ]
 
 # Build the tarballs, and possibly a `build.jl` as well.
-build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"8", julia_compat="1.6")
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies; preferred_gcc_version=v"11", julia_compat="1.6")
```
Comment (giordano) [inline G/Geant4/build_tarballs.jl:112]: GCC 11 can be problematic for Julia v1.6, for C++ code: https://github.com/JuliaPackaging/Yggdrasil/blob/9e93401baf07c15c00d239fc6c4d124f8121451f/CONTRIBUTING.md#compatibility-tips.  Would GCC 10 be sufficient?

Comment (peremato) [reply G/Geant4/build_tarballs.jl:112]: Should be also OK I guess. I'll run a test

File: G/Geant4/bundled/patches/windows.patch
Comment (peremato) [reply G/Geant4/bundled/patches/windows.patch:1]: Yes of course. I was planing on it.

File: G/Geant4/build_tarballs.jl (hunk 4 above)
Comment (peremato) [reply G/Geant4/build_tarballs.jl]: Done. The release notes says that has also been tested with gcc10

File: G/Geant4/bundled/patches/windows.patch (hunk 3 above)
Comment (peremato) [reply G/Geant4/bundled/patches/windows.patch]: Done

**Pull request merged**